storage/range.go

```diff
@@ -1575,13 +1581,12 @@ func (r *Range) loadFirstAndLastIndex() error {
 			return err
 		}
 		atomic.StoreUint64(&r.lastIndex, lastEnt.Index)
-
-		var firstEnt raftpb.Entry
-		err = gogoproto.Unmarshal(kvs[len(kvs)-1].Value.Bytes, &firstEnt)
+	} else if len(kvs) == 0 {
+		ts, err := r.raftTruncatedState()
 		if err != nil {
-			return err
+			return nil
```

Comment:
Looks like we need to do something more specific here and also document what's going on a little better with this method.